gm4_lively_lily_pads/data/gm4_lively_lily_pads/function/mechanics/right_click_detection/process.mcfunction

```diff
@@ -0,0 +1,23 @@
+# processes the "right click detection" interaction so that it is despawned when unneeded
+# @s = gm4_llp_placement_rcd interaction 
+# at @s
+# run from mechanics/right_click_detection/loop
+
+# visuals
+particle scrape ~ ~.1 ~ 0 0 0 0 1
+
+# refresh timer
+scoreboard players set $timer gm4_llp.data 10
+
+# kill rcd if uneeded
+execute unless entity @a[tag=gm4_llp_holding_item,distance=..6] run return run kill @s
+execute unless block ~ ~ ~ lily_pad run return run kill @s
+execute if entity @e[type=block_display,tag=gm4_llp_display,distance=..0.1] run return run kill @s
```

Comment:
Here I'd prefer checking the entire blockspace just so we don't get into trouble when changing the intra-block-position of this entity at some point.
```suggestion
execute align xyz if entity @e[type=block_display,tag=gm4_llp_display,dx=0] run return run kill @s
```